Chessington.GameEngine/Pieces/Piece.cs

```diff
@@ -20,5 +20,46 @@ public void MoveTo(Board board, Square newSquare)
             var currentSquare = board.FindPiece(this);
             board.MovePiece(currentSquare, newSquare);
         }
+
+        protected IEnumerable<Square> GetAvailableLateralMoves(Board board)
+        {
+            var currentSquare = board.FindPiece(this);
+
+            var zeroes = Enumerable.Repeat(0, 15);
+            var minusSevenToSeven = Enumerable.Range(-7, 15);
+
+            var moveVectors = zeroes.Zip(minusSevenToSeven, (x, y) => new MoveVector(x,y));
+            moveVectors = moveVectors.Union(minusSevenToSeven.Zip(zeroes, (x, y) => new MoveVector(x, y)));
+
+            return MoveVector.FindAvailableMovesFromVectors(moveVectors, board, currentSquare);
+        }
+
+        protected IEnumerable<Square> GetAvailableDiagonalMoves(Board board)
+        {
+            var currentSquare = board.FindPiece(this);
+
+            // Produce a zipped set of diagonal directions (South West, South East, North West, North East)
+            var diagonalDirections = new List<MoveVector>(new[]
+            {
+                new MoveVector(-1, -1),
+                new MoveVector(-1, 1),
+                new MoveVector(1, -1),
+                new MoveVector(1, 1)
+            });
+
+            // Send diagonal lines out 8 squares in each diagonal direction
+            var moveVectors = diagonalDirections.Select(vector =>
+            {
+                return Enumerable.Range(1, 7).Select(i => new MoveVector(
+                    vector.RowChange * i,
+                    vector.ColChange * i)
+                );
+            });
+
+            // Flatten from four lists of lists of squares to one list
+            var moveVectorsFlattened = moveVectors.SelectMany(x => x).ToList();
```

Comment:
A select inside a select followed by a select many can also be a bit tricky to read. What about something like:
```
var oneToSeven = Enumerable.Range(1, 7);
var bothPositiveDiagonal = oneToSeven.Select(x => new MoveVector(x, x));
var xNegativeDiagonal = oneToSeven.Select(x => new MoveVector(-x, x));
var yNegativeDiagonal = oneToSeven.Select(x => new MoveVector(x, -x));
var bothNegativeDiagonal = oneToSeven.Select(x => new MoveVector(-x, -x));
var allDiagonalMoves = bothPositiveDiagonal
  .Concat(xNegativeDiagonal)
  .Concat(yNegativeDiagonal)
  .Concat(bothNegativeDiagonal);
```
🙂

Reply:
Yeah for sure. Another case of me unnecessarily avoiding hard-coding for the sake of it and at the cost of readability. Sorted in fbf9e81